File: javascript/lib/operation.js
import fs from 'fs';

const inputPath = new URL('../input.json', import.meta.url);
let Operation = [{operation:'', unitCost: 0, quantity: 0, assetClass: []}];

try {
    const raw = JSON.parse(fs.readFileSync(inputPath, 'utf8'));
    if (Array.isArray(raw.operations)) Operation = raw.operations;

    
} catch (e) {

    Operation = [];
}

function executeOperation (){
    console.log('Operation executed');
}

export { Operation, executeOperation };

File: python/models/operation.py
class OperationType:
    BUY = "buy"
    SELL = "sell"

class Operation:
    def __init__(self, operation: OperationType, unitCost: int = 0, quantity: int = 0, assetClass: list | None = None):
        self.operation = operation
        self.unitCost = unitCost
        self.quantity = quantity
        self.assetClass = assetClass or []